### PR TITLE
Add CgroupInfo API for container resource monitoring

### PR DESCRIPTION
**Pull request by rohan-coder02**
## Summary

Implements a new `CgroupInfo` interface to expose cgroup resource limits and usage metrics, enabling OSHI users to detect containerized environments and retrieve resource constraints.

Relates to #3224

## Usage Examples

```java
SystemInfo si = new SystemInfo();
OperatingSystem os = si.getOperatingSystem();
CgroupInfo cgroup = os.getCgroupInfo();

// Check if running in a container
if (cgroup.isContainerized()) {
    System.out.println("Running in cgroup v" + cgroup.getVersion());
    
    // CPU limits
    double effectiveCpus = cgroup.getEffectiveCpus();
    if (effectiveCpus > 0) {
        System.out.println("CPU limit: " + effectiveCpus + " cores");
    }
    
    // Memory limits
    long memLimit = cgroup.getMemoryLimit();
    if (memLimit < CgroupInfo.UNLIMITED_MEMORY) {
        System.out.println("Memory limit: " + memLimit / (1024 * 1024) + " MB");
    }
    System.out.println("Memory usage: " + cgroup.getMemoryUsage() / (1024 * 1024) + " MB");
    
    // PID limits
    long pidLimit = cgroup.getPidLimit();
    if (pidLimit != CgroupInfo.UNLIMITED) {
        System.out.println("PID limit: " + pidLimit);
    }
    System.out.println("Current PIDs: " + cgroup.getPidCurrent());
}
```

### `CgroupInfo` interface

- All methods have `default` implementations returning sentinel values, so non-Linux platforms get correct no-op behavior without a separate class
- Named constants for sentinel values: `UNLIMITED` (-1), `DEFAULT_CPU_PERIOD` (100000), `UNLIMITED_MEMORY` (Long.MAX_VALUE), `UNLIMITED_CPUS` (-1.0d)
- `getEffectiveCpus()` is a default method computed from `getCpuQuota() / getCpuPeriod()`
- `OperatingSystem.getCgroupInfo()` returns an anonymous instance using the defaults

### `LinuxCgroupInfo` implementation

- **Version detection**: Reads `/proc/filesystems` for cgroup2 support
- **v2 support** (primary): Parses `cpu.max`, `cpu.stat`, `memory.max`, `memory.current`, `pids.max`, `pids.current`
- **v1 fallback**: Parses controller-specific files (`cpu.cfs_quota_us`, `cpuacct.usage`, etc.)
- **Memoization**: Version and cgroup path cached permanently; limit values (quota, period, memory, PIDs) use `defaultExpiration()` TTL since they can be reconfigured at runtime
- **Testability**: V2 read methods have package-private overloads accepting a base path, enabling unit tests with `@TempDir`

### Other changes

- Version bumped to 7.1.0-SNAPSHOT (new public API)
- `ProcPath.java` — added `SELF_CGROUP` and `FILESYSTEMS` constants
- `SysPath.java` — added `CGROUP` constant
- `LinuxOperatingSystem.java` — overrides `getCgroupInfo()` with memoized instance

## Summary by CodeRabbit

* **New Features**
  * Added CgroupInfo API and Linux cgroup support (v1/v2): reports container status, cgroup version, CPU quota/period/usage, effective CPUs, memory limit/usage, and PID limits/current. OS integration surfaces cgroup info in system output.

* **Tests**
  * Added unit tests and gated container integration tests covering v1/v2, edge cases, and defaults.

* **Chores**
  * Bumped project version to 7.1.0, updated changelog, and added CI job for cgroup integration tests.

### CODE REVIEW

**Comment by dbwiddis** [comment]
The lint failure is because we added to the API which requires a minor version bump to 7.1.0 :)

**Comment by dbwiddis** [comment]
Implementation looks good. We need some tests!

File: oshi-common/src/main/java/oshi/software/common/os/linux/LinuxCgroupInfo.java [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+
+    private static final String[] CONTAINER_MARKERS = { "/docker/", "/kubepods/", "/lxc/", "/containerd/", "/crio-",
+            "/buildkit/" };
+
+    private final Supplier<Integer> versionSupplier = Memoizer.memoize(this::detectVersion);
+    private final Supplier<String> cgroupPathSupplier = Memoizer.memoize(this::parseCgroupPath);
+    private final Supplier<Long> cpuQuotaSupplier = Memoizer.memoize(this::readCpuQuota);
+    private final Supplier<Long> cpuPeriodSupplier = Memoizer.memoize(this::readCpuPeriod);
+    private final Supplier<Long> memoryLimitSupplier = Memoizer.memoize(this::readMemoryLimit);
+    private final Supplier<Long> pidLimitSupplier = Memoizer.memoize(this::readPidLimit);
```
Comment (dbwiddis) [inline oshi-common/src/main/java/oshi/software/common/os/linux/LinuxCgroupInfo.java]: While the version and path won't change, I think it's possible to change some of the other parameters during the container's operation.  So these need some sort of expiration.

File: oshi-common/src/main/java/oshi/software/common/os/linux/LinuxCgroupInfo.java [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+    }
+
+    private long readCpuPeriod() {
+        int version = getVersion();
+        if (version == 2) {
+            return readCpuPeriodV2();
+        } else if (version == 1) {
+            return readCpuPeriodV1();
+        }
+        return 100000L;
```
Comment (dbwiddis) [inline oshi-common/src/main/java/oshi/software/common/os/linux/LinuxCgroupInfo.java]: We're returning this magic number a lot. We should make it a constant (`DEFAULT_CPU_PERIOD` perhaps?) and we could reference that constant in the interface javadoc.

Similar comment applies to the other sentinel values (-1L, -1d) that are not self explanatory (0, empty string, max value).

File: oshi-common/src/main/java/oshi/software/common/os/linux/LinuxCgroupInfo.java [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+        }
+        String[] parts = cpuMax.split("\\s+");
+        if (parts.length >= 1) {
+            if ("max".equalsIgnoreCase(parts[0])) {
+                return -1L;
+            }
+            return ParseUtil.parseLongOrDefault(parts[0], -1L);
+        }
+        return -1L;
+    }
```
Comment (dbwiddis) [inline oshi-common/src/main/java/oshi/software/common/os/linux/LinuxCgroupInfo.java:257]: Let's make all the methods like these testable by having the no-arg  version delegate to a package-private version taking a string path value that we can then test.

So here we would have 
```java
    private long readCpuQuotaV2() {
        return readCpuQuotaV2(getV2CGroupBase());
    }
```
and then
```java
    // package private for testing
    long readCpuQuotaV2(String basePath) {
        // the existing method
    }
```

Then you could test them by passing strings from test files.

See the pattern here:
https://github.com/oshi/oshi/blob/459e2242eb672cbf6a492547022105249ccc504d/oshi-common/src/main/java/oshi/hardware/common/platform/linux/LinuxCentralProcessor.java#L505-L509

Tested like this:
https://github.com/oshi/oshi/blob/459e2242eb672cbf6a492547022105249ccc504d/oshi-common/src/test/java/oshi/hardware/common/platform/linux/LinuxCentralProcessorTest.java#L145-L158

File: oshi-common/src/main/java/oshi/software/common/NoOpCgroupInfo.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    @Override
+    public long getCpuQuota() {
+        return -1L;
+    }
+
+    @Override
+    public long getCpuPeriod() {
+        return 100000L;
```
Comment (dbwiddis) [inline oshi-common/src/main/java/oshi/software/common/NoOpCgroupInfo.java]: This class might be a great place to define these constants.

Alternately instead of `NoOpCgroupInfo` could call it `DefaultCgroupInfo` or just `AbstractCgroupInfo` and inherit from it and then your default return values could be `super.getCpuPeriod()`.

Alternately define the sentinel values on the interface as `default` and skip this class entirely.
